src/components/modal/product-modal.tsx

```diff
@@ -1,52 +1,58 @@
-// import React from "react";
-import { useDispatch } from "react-redux";
-import { productModalDisplay } from "../../slices/modal";
+import { useDispatch, useSelector } from "react-redux";
+import { closeModal } from "../../slices/modal";
+import { getActiveProduct } from "../../slices/products";
 import './modal.scss';
 import ImageCard from "../image-card/image-card";
+import { RootState } from "../../store";
 
 const ProductModal = () => {
-    const dispatch = useDispatch();
-    const closeModal = (e:any) => {
-        e.stopPropagation();
-        dispatch(productModalDisplay())
+    const shoeView = useSelector((state:RootState) => state.product.showView)
+    const dispatch = useDispatch(); 
+    
+    const getActiveItem = (shop: typeof shoeView) => {
+        let itemIndex;
+        for(const sh in shop){
+            const item = shop[sh]
+            if(item.active === true){
+                itemIndex = sh;
+            }
+        }
+        return itemIndex;
```

Comment:
Did typescript complain about the type of itemIndex? @onyi2020

Reply:
Nope.